### PR TITLE
LegacyStorageFileRowReader: handle the case when statement->fetch returns false

### PR DESCRIPTION
**Pull request by SerheyDolgushev**
| Question                                  | Answer
| ---------------------------------------- | ------------------
| **JIRA issue**                          | [EZP-31748](https://jira.ez.no/browse/EZP-31748)
| **Type**                                   | bug
| **Target eZ Platform version** | `v3.0.5`, `v3.1.0`
| **BC breaks**                          | no
| **Tests pass**                          | yes
| **Doc needed**                       | no

In some cases `ezplatform:io:migrate-files` command fails because of the following error:
```bash
bin/console ezplatform:io:migrate-files --from=default,default --to=XXX,YYY --no-interaction -vvv

...

In LegacyStorageFileRowReader.php line 45:
 
 [ErrorException] 
 Notice: Trying to access array offset on value of type bool

```
#### Checklist:
- [x] PR description is updated.
- [x] Tests are implemented.
- [x] Added code follows Coding Standards (use `$ composer fix-cs`).
- [x] PR is ready for a review.

### CODE REVIEW

**Review by kaff — APPROVED** [review]

File: eZ/Bundle/EzPublishIOBundle/Migration/FileLister/FileRowReader/LegacyStorageFileRowReader.php [hunk 1]
```diff
@@ -42,7 +42,7 @@ public function __construct(Connection $connection)
     {
         $row = $this->statement->fetch();
 
-        return $this->prependMimeToPath($row['filename'], $row['mime_type']);
+        return is_array($row) ? $this->prependMimeToPath($row['filename'], $row['mime_type']) : null;
```
Comment (alongosz) [inline eZ/Bundle/EzPublishIOBundle/Migration/FileLister/FileRowReader/LegacyStorageFileRowReader.php:45]: This should be more strict, the fetch result can be either array or boolean, so this should be enough:

```suggestion
        return false !== $row ? $this->prependMimeToPath($row['filename'], $row['mime_type']) : null;
```

**Review by alongosz — CHANGES_REQUESTED** [review]

**Comment by SerheyDolgushev** [comment]
@alongosz committed your suggestion

**Review by alongosz — APPROVED** [review]

**Comment by lserwatka** [comment]
@alongosz could you merge it up?

**Comment by alongosz** [comment]
Merged up to master.

Thank you @SerheyDolgushev :tada:

**Pull request merged**